sdk/servicebus/microsoft-azure-servicebus/src/main/java/com/microsoft/azure/servicebus/management/ManagementClientAsync.java

```diff
@@ -92,9 +92,48 @@ public ManagementClientAsync(URI namespaceEndpointURI, ClientSettings clientSett
         DefaultAsyncHttpClientConfig.Builder clientBuilder = Dsl.config()
                 .setConnectTimeout((int) CONNECTION_TIMEOUT.toMillis())
                 .setRequestTimeout((int) this.clientSettings.getOperationTimeout().toMillis());
+
+        if(shouldUseProxy(this.namespaceEndpointURI.getHost())){
+            InetSocketAddress address = (InetSocketAddress)this.proxies.get(0).address();
+            String proxyHostName=address.getHostName();
+            int proxyPort=address.getPort();
+
+            clientBuilder.setProxyServer(new ProxyServer.Builder(proxyHostName,proxyPort));
+        }
+
         this.asyncHttpClient = asyncHttpClient(clientBuilder);
     }
 
+    public boolean shouldUseProxy(final String hostName) {
```

Comment:
These three methods are taken out from ProxyConnectionHandler. Will leave as a TODO to make them as a util class and shared between these two classes.